src/ncp/ncp_base_ftd.cpp

```diff
@@ -1438,6 +1438,34 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_INFRA_IF_STATE>(void)
     return error;
 }
 
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_INFRA_IF_RECV_ICMP6>(void)
+{
+    otError                 error = OT_ERROR_NONE;
+    spinel_ipv6_icmp_type_t msgType;
+    uint32_t                infraIfIndex;
+    const otIp6Address     *address;
+    const uint8_t          *icmp6Data = nullptr;
+    uint16_t                len;
+
+    SuccessOrExit(error = mDecoder.ReadUint32(infraIfIndex));
+    VerifyOrExit(mInfraIfIndex == infraIfIndex, error = OT_ERROR_DROP);
+    SuccessOrExit(error = mDecoder.ReadUint8(msgType));
+    SuccessOrExit(error = mDecoder.ReadIp6Address(address));
+    SuccessOrExit(error = mDecoder.ReadData(icmp6Data, len));
+
+    if (msgType == SPINEL_IPV6_ICMP_TYPE_ND)
+    {
+        otPlatInfraIfRecvIcmp6Nd(mInstance, infraIfIndex, address, icmp6Data, len);
```

Comment:
Add guard for `#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE` around this?

I think otherwise the `otPlatInfraIfRecvIcmp6Nd` is not provided.

Reply:
Good catch! Have updated the guard to `#if OPENTHREAD_CONFIG_NCP_INFRA_IF_ENABLE && OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE` for the whole block. The previous functions call `otBorderRoutingXXX` which also needs to be guarded by this.